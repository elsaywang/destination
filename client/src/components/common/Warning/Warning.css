:local(.warning) {
    display: flex;
    margin-left: 0.5em;
    padding: 0.15em;
    color: var(--spectrum-semantic-notice-color-status);
    justify-content: flex-end;
}
:local(.warningMessage) {
    margin-left: 0.5em;
    font-size: var(--spectrum-global-font-size-medium);
    font-weight: var(--spectrum-global-font-weight-regular);
    font-style: italic;
}
