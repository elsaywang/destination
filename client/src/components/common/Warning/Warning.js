import React from 'react';
import styles from './Warning.css';
import Alert from '@react/react-spectrum/Icon/Alert';

const Warning = props => {
    return (
        <div className={styles.warning}>
            <Alert size="S" variant="warning" />
            <div className={styles.warningMessage}>{props.children}</div>
        </div>
    );
};

export default Warning;
