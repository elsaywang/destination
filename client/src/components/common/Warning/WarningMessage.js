import React from 'react';
import PropTypes from 'prop-types';

const WarningMessage = ({ content }) => <span>{content}</span>;

WarningMessage.propTypes = {
    content: PropTypes.string,
};
export default WarningMessage;
