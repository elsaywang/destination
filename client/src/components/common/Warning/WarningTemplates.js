import React from 'react';
import PropTypes from 'prop-types';

export const SignalsSelectionWarningMessage = () => (
    <span>
        We canâ€™t create traits from signals of different types. Try again using signals all from
        either <strong> Real-Time Signals </strong> or <strong> Onboarded Signals</strong>.
    </span>
);

export const SignalsSelectionLimitMessage = ({ maxSignalSelections }) => (
    <span>
        You may select up to <strong>{maxSignalSelections}</strong> Signals. Only the first{' '}
        <strong>{maxSignalSelections}</strong> selected signals will be used to create a trait.
    </span>
);

SignalsSelectionLimitMessage.propTypes = {
    maxSignalSelections: PropTypes.number.isRequired,
};
