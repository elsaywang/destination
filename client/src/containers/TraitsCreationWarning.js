import React, { Component, Fragment } from 'react';
import PropTypes from 'prop-types';
import Warning from '../components/common/Warning';
import {
    SignalsSelectionWarningMessage,
    SignalsSelectionLimitMessage,
} from '../components/common/Warning/WarningTemplates';
import {
    isMaxSignalSelectionsReached,
    hasSignalSelectionsTypeWarning,
    getMaxSignalSelections,
} from '../reducers/selectedSignals';
import { connect } from 'react-redux';

export class TraitsCreationWarning extends Component {
    render() {
        const {
            hasSignalSelectionsTypeWarning,
            isMaxSignalSelectionsReached,
            maxSignalSelections,
        } = this.props;

        if (hasSignalSelectionsTypeWarning && isMaxSignalSelectionsReached) {
            return (
                <Fragment>
                    <Warning>
                        <SignalsSelectionLimitMessage maxSignalSelections={maxSignalSelections} />
                    </Warning>
                    <Warning>
                        <SignalsSelectionWarningMessage />
                    </Warning>
                </Fragment>
            );
        } else if (hasSignalSelectionsTypeWarning) {
            return (
                <Warning>
                    <SignalsSelectionWarningMessage />
                </Warning>
            );
        } else if (isMaxSignalSelectionsReached) {
            return (
                <Warning>
                    <SignalsSelectionLimitMessage maxSignalSelections={maxSignalSelections} />
                </Warning>
            );
        }
        return null;
    }
}

TraitsCreationWarning.propTypes = {
    hasSignalSelectionsTypeWarning: PropTypes.bool,
    isMaxSignalSelectionsReached: PropTypes.bool,
    maxSignalSelections: PropTypes.number,
};

const mapStateToProps = ({ selectedSignals }) => ({
    isMaxSignalSelectionsReached: isMaxSignalSelectionsReached(selectedSignals),
    hasSignalSelectionsTypeWarning: hasSignalSelectionsTypeWarning(selectedSignals),
    maxSignalSelections: getMaxSignalSelections(selectedSignals),
});

export default connect(
    mapStateToProps,
    null,
)(TraitsCreationWarning);
