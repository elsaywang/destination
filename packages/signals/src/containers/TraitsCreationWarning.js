import React, { Component } from 'react';
import PropTypes from 'prop-types';
import Warning from '../components/common/Warning';
import {
    SignalsSelectionWarningMessage,
    SignalsSelectionLimitMessage,
    OnboardedSignalSelectionWarningMessage,
} from '../components/common/Warning/WarningTemplates';
import {
    isMaxSignalSelectionsReached,
    hasSignalSelectionsTypeWarning,
    getMaxSignalSelections,
} from '../reducers/selectedSignals';
import { invalidSelectedOnboardedResultsTraitCreation } from '../reducers';
import { connect } from 'react-redux';

export class TraitsCreationWarning extends Component {
    render() {
        const {
            hasSignalSelectionsTypeWarning,
            isMaxSignalSelectionsReached,
            maxSignalSelections,
            hasOnboardedSignalSelectionsWarning,
        } = this.props;

        if (isMaxSignalSelectionsReached) {
            return (
                <Warning dataTest="multi-signals-trait-creation-warning-max-selections">
                    <SignalsSelectionLimitMessage maxSignalSelections={maxSignalSelections} />
                </Warning>
            );
        } else if (hasSignalSelectionsTypeWarning) {
            return (
                <Warning dataTest="multi-signals-trait-creation-warning-selections-type">
                    <SignalsSelectionWarningMessage />
                </Warning>
            );
        } else if (hasOnboardedSignalSelectionsWarning) {
            return (
                <Warning dataTest="multi-signals-trait-creation-warning-onboarded-selections">
                    <OnboardedSignalSelectionWarningMessage />
                </Warning>
            );
        }
        return null;
    }
}

TraitsCreationWarning.propTypes = {
    hasSignalSelectionsTypeWarning: PropTypes.bool,
    hasOnboardedSignalSelectionsWarning: PropTypes.bool,
    isMaxSignalSelectionsReached: PropTypes.bool,
    maxSignalSelections: PropTypes.number,
};

const mapStateToProps = ({ selectedSignals, results }) => ({
    isMaxSignalSelectionsReached: isMaxSignalSelectionsReached(selectedSignals),
    hasSignalSelectionsTypeWarning: hasSignalSelectionsTypeWarning(selectedSignals),
    hasOnboardedSignalSelectionsWarning: invalidSelectedOnboardedResultsTraitCreation({
        selectedSignals,
        results,
    }),
    maxSignalSelections: getMaxSignalSelections(selectedSignals),
});

export default connect(
    mapStateToProps,
    null,
)(TraitsCreationWarning);
