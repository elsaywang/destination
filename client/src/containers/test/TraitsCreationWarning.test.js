import React from 'react';
import { shallow } from 'enzyme';
import { TraitsCreationWarning } from '../TraitsCreationWarning';
import Warning from '../../components/common/Warning';
import { SignalsSelectionWarningMessage } from '../../components/common/Warning/WarningTemplates';

describe('<TraitsCreationWarning/> component', () => {
    describe('rendering when there is no signal selected', () => {
        const initialprops = {
            selectedSignals: {
                selectionMessage: '',
                records: [],
                hasWarning: false,
            },
        };
        const { hasWarning } = initialprops.selectedSignals;
        const wrapper = shallow(<TraitsCreationWarning hasWarning={hasWarning} />);

        it('matches snapshot', () => {
            expect(wrapper).toMatchSnapshot();
        });

        it('renders null element', () => {
            expect(wrapper.getElement()).toBe(null);
        });
    });

    describe('rendering when there are signals selected but without warning', () => {
        const props = {
            selectedSignals: {
                selectionMessage: '1 Real-time signal selected ',
                records: [{ rowIndex: 0, signalType: 'Adobe Analytics' }],
                hasWarning: false,
            },
        };
        const { hasWarning } = props.selectedSignals;
        const wrapper = shallow(<TraitsCreationWarning hasWarning={hasWarning} />);

        it('matches snapshot', () => {
            expect(wrapper).toMatchSnapshot();
        });

        it('renders null element', () => {
            expect(wrapper.getElement()).toBe(null);
        });
    });

    describe('rendering when there are signals selected with warning', () => {
        const props = {
            selectedSignals: {
                selectionMessage: '1 Onboarded , 1 Real-time are selected',
                records: [
                    { rowIndex: 0, signalType: 'Adobe Analytics' },
                    { rowIndex: 4, signalType: 'Onboarded Records' },
                ],
                hasWarning: true,
            },
        };
        const { hasWarning } = props.selectedSignals;
        const wrapper = shallow(<TraitsCreationWarning hasWarning={hasWarning} />);

        it('matches snapshot', () => {
            expect(wrapper).toMatchSnapshot();
        });

        it('renders <Warning/> component with <SignalsSelectionWarningMessage/> template', () => {
            expect(wrapper.find(Warning).exists()).toBe(true);
            expect(wrapper.find(SignalsSelectionWarningMessage).exists()).toBe(true);
        });
    });
});
