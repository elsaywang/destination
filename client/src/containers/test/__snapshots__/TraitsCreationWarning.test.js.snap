// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TraitsCreationWarning/> component rendering when \`hasSignalSelectionsTypeWarning\` and \`isMaxSignalSelectionsReached\` props both are false matches snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TraitsCreationWarning
    hasSignalSelectionsTypeWarning={false}
    isMaxSignalSelectionsReached={false}
    maxSignalSelections={5}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): null,
  Symbol(enzyme.__nodes__): Array [
    null,
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getSnapshotBeforeUpdate": true,
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;

exports[`<TraitsCreationWarning/> component rendering when \`hasSignalSelectionsTypeWarning\` prop is true matches snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TraitsCreationWarning
    hasSignalSelectionsTypeWarning={true}
    isMaxSignalSelectionsReached={false}
    maxSignalSelections={5}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": <Unknown />,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {},
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <Unknown />,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getSnapshotBeforeUpdate": true,
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;

exports[`<TraitsCreationWarning/> component rendering when \`isMaxSignalSelectionsReached\` props is true matches snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TraitsCreationWarning
    hasSignalSelectionsTypeWarning={false}
    isMaxSignalSelectionsReached={true}
    maxSignalSelections={5}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": <Unknown
        maxSignalSelections={5}
/>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "maxSignalSelections": 5,
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <Unknown
          maxSignalSelections={5}
/>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "maxSignalSelections": 5,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getSnapshotBeforeUpdate": true,
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
